### PR TITLE
plan: keep error message consistent with MySQL (#3962)

### PR DESCRIPTION
**Pull request by jackysp**
Fix issue #3962 @hicqu @XuHuaiyu

### CODE REVIEW

File: plan/resolver.go [hunk 1]
```diff
@@ -24,6 +24,18 @@ import (
 	"github.com/pingcap/tidb/mysql"
 	"github.com/pingcap/tidb/table"
 	"github.com/pingcap/tidb/util/types"
+	"strings"
```
Comment (tiancaiamao) [inline plan/resolver.go]: Put it before any third-party packages.

File: plan/resolver.go [hunk 2]
```diff
@@ -512,18 +527,23 @@ func (nr *nameResolver) handleColumnName(cn *ast.ColumnNameExpr) {
 			return
 		}
 	}
-	nr.Err = errors.Errorf("unknown column %s", cn.Name.Name.L)
+	fieldName := cn.Name.Name.String()
+	if !strings.EqualFold(cn.Name.Table.String(), "") {
```
Comment (tiancaiamao) [inline plan/resolver.go]: `if len(cn.Name.Table.String()) != 0`

File: plan/resolver.go [hunk 3]
```diff
@@ -611,7 +631,12 @@ func (nr *nameResolver) resolveColumnNameInOnCondition(cn *ast.ColumnNameExpr) {
 	join := ctx.joinNodeStack[len(ctx.joinNodeStack)-1]
 	tableSources := appendTableSources(nil, join)
 	if !nr.resolveColumnInTableSources(cn, tableSources) {
-		nr.Err = errors.Errorf("unknown column name %s", cn.Name.Name.O)
+		fieldName := cn.Name.Name.String()
+		if !strings.EqualFold(cn.Name.Table.String(), "") {
```
Comment (tiancaiamao) [inline plan/resolver.go]: ditto

**Comment by jackysp** [comment]
@tiancaiamao PTAL

**Comment by tiancaiamao** [comment]
LGTM

**Comment by shenli** [comment]
LGTM

**Pull request merged**